packages/core/http/core-http-router-server-internal/src/versioned_router/is_valid_route_version.ts

```diff
@@ -5,11 +5,16 @@
  * in compliance with, at your election, the Elastic License 2.0 or the Server
  * Side Public License, v 1.
  */
-
 import moment from 'moment';
 
 const PUBLIC_VERSION_REGEX = /^[0-9]{4}-[0-9]{2}-[0-9]{2}$/;
-const INTERNAL_VERSION_REGEX = /^[0-9]+$/;
+const INTERNAL_VERSION_REGEX = /^[1-9][0-9]*$/;
+
+export function isAllowedPublicVersion(version: string): undefined | string {
+  if ('2023-10-31' !== version) {
+    return `Invalid public version, for now please use "2023-10-31" as the version for all public routes. Received "${version}".}"`;
+  }
```

Comment:
Please at least use a constant for this 🙈 

Also, if it's the only allowed version for now, we may want to expose it from a public package for teams to import/use the const?

Reply:
> Please at least use a constant for this 🙈

Sure

> Also, if it's the only allowed version for now

I'd rather not expose this as a const since this date is somewhat arbitrary and we are only pretending that everyone chose the same date. In practice API authors can independently introduce new public versions down the line, then this code should be deleted.